libraries/primitive_consul_maintenance.rb

```diff
@@ -23,15 +23,30 @@ def register(choregraphie)
 
       choregraphie.before do
         require 'diplomat'
-        Diplomat::Maintenance.enable(true, @options[:reason])
+        checks = Diplomat::Agent.checks()
+        maint_status = checks.dig('_node_maintenance', 'Status')
+        if maint_status == 'critical'
+          maint_notes = checks.dig('_node_maintenance', 'Notes')
+          Chef::Log.warn "Consul maintenance was already enabled, reason: #{maint_notes}."
+        else
+          Diplomat::Maintenance.enable(true, @options[:reason])
+        end
       end
 
       choregraphie.cleanup do
         require 'diplomat'
-        # WARNING
-        # This means that any chef run will disable the maintenance mode on the
-        # node
-        Diplomat::Maintenance.enable(false, @options[:reason])
+        checks = Diplomat::Agent.checks()
+        maint_status = checks.dig('_node_maintenance', 'Status')
+        if maint_status == 'critical'
+          maint_notes = checks.dig('_node_maintenance', 'Notes')
+          Chef::Log.info "Consul maintenance is enabled, reason: #{maint_notes}."
+          if maint_notes == @options[:reason]
+            Chef::Log.info "Consul maintenance will be disabled."
+            Diplomat::Maintenance.enable(false, @options[:reason])
+          else
+            Chef::Log.warn "Consul maintenance won't be disabled because the following reason was expected: #{@options[:reason]}."
```

Comment:
I would prefer to see the current maintenance reason logged instead of the reason that would have allowed this primitive to remove it.
Maybe a message like: `Custom maintenance already activated "#{maint_notes}" by something else than this choregraphie. We won't disable it.`

Reply:
It is already logged in the previous message. But I will change to have all information in the warn log.